### PR TITLE
Decrease build warnings

### PR DESCRIPTION
**Pull request by livecodefraser**
Reduces the number of build warnings we get by a combination of fixing some issues in headers and disabling the warnings that cause lots of noise without being particularly helpful.

---

This change is [Reviewable](https://reviewable.io/reviews/livecode/livecode/3964)

### CODE REVIEW

**Comment by runrevmark** [comment]
Reviewed 28 of 28 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/livecode/livecode/3964)_

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 1599da0

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 1599da0c8a8ca36a2775abef71b19920e4b4c948

**Comment by livecode-vulcan** [comment]
😎 test success 1599da0c8a8ca36a2775abef71b19920e4b4c948
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2445)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2455)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2438)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2456)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2462)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2466)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2478)

**Comment by livecodefraser** [comment]
26,130 warnings -> 9,259 warnings
(iOS 9.3 device build)

File: engine/src/system.h [hunk 1]
```diff
@@ -188,7 +188,7 @@ class MCMemoryFileHandle: public MCSystemFileHandle
 		// If there isn't enough room, extend
 		if (m_pointer + p_length > m_capacity)
```
Comment (mwieder) [inline engine/src/system.h:189]: interesting, though, that we're still mixing uint32_t and size_t variables several places in this file.

Comment (runrevmark) [inline engine/src/system.h:189]: Not particularly. C does automatic upward promotion of integer types (as makes perfect sense) and warnings come about (generally) when you are casting a larger integer type to a smaller one (e.g. signed -> unsigned, int64 -> int32). The above line is just a comparison, no result is stored anywhere so auto promotion is absolutely fine.

Comment (peter-b) [inline engine/src/system.h:189]: Ooh, I just noticed that this appears to be missing an overflow check when calculating the new capacity.

**Pull request merged**